src/Racing.cpp

```diff
@@ -14,35 +12,125 @@ std::map<int32_t, EPInfo> Racing::EPData;
 std::map<CNSocket*, EPRace> Racing::EPRaces;
 std::map<int32_t, std::pair<std::vector<int>, std::vector<int>>> Racing::EPRewards;
 
-static void racingStart(CNSocket* sock, CNPacketData* data) {
-    auto req = (sP_CL2FE_REQ_EP_RACE_START*)data->buf;
+namespace {
+    // Helper functions
+    bool validateRaceStart(CNSocket* sock, int32_t startEcomID) {
+        if (NPCManager::NPCs.find(startEcomID) == NPCManager::NPCs.end())
+            return false; // starting line agent not found
 
-    if (NPCManager::NPCs.find(req->iStartEcomID) == NPCManager::NPCs.end())
-        return; // starting line agent not found
+        int mapNum = MAPNUM(NPCManager::NPCs[startEcomID]->instanceID);
+        return EPData.find(mapNum) != EPData.end() && EPData[mapNum].EPID != 0;
+    }
```

Comment:
Should validate that a player is not already in a race.
I would also recommend having an output argument for mapnum to avoid having to recalculate it in the packet handlers.